scripts/git_page/md_post_process.py

```diff
@@ -0,0 +1,191 @@
+
+
+def workflow_md_has_multiple_subheaders(md_path: str) -> bool:
+    """
+    Checks if the markdown file has multiple subheaders
+    :param md_path: Path to markdown file
+    :return: True if the markdown file has multiple subheaders
+    """
+    with open(md_path, 'r') as md_file:
+        md_lines = md_file.readlines()
+        if len(md_lines) == 0:
+            return False
+
+        subheader_count = 0
+        for line in md_lines:
+            if line.startswith('## '):
+                subheader_count += 1
+
+        return subheader_count > 1
+
+
+def add_task_subheader(md_path: Union[str, Path]) -> None:
+    """
+    Adds a task subheader to the markdown file after the first subheader.
+    :param md_path: Path to markdown file
+    :return: None
+    """
+    try:
+        with open(md_path, 'r+') as md_file:
+            md_lines = md_file.readlines()
+
+            try:
+                second_subhead_index = [i for i, line in enumerate(md_lines) if line.startswith("## ")][1]
+            except IndexError:
+                raise IndexError(f"Unable to find second subheader in {md_path}")
+
+            for i, line in enumerate(md_lines[second_subhead_index:]):
+                if line.startswith("#"):
+                    md_lines[second_subhead_index + i] = "#" + line
+
+            task_header = "## Tasks\n\n"
+            md_lines.insert(second_subhead_index, task_header)
+
+            md_file.seek(0, 0)
+            md_file.writelines(md_lines)
+
+    except (FileNotFoundError, IsADirectoryError):
+        Logger.debug(f"Unable to open file at {md_path}")
+
+    Logger.debug(f'Added task subheader to {md_path}')
+
+
+def remove_subheaders_after_first(md_path: Union[str, Path]) -> None:
+    """
+    Removes the subheader from the markdown file after the first subheader.
+    :param md_path: Path to markdown file
+    :return: None
+    """
+    try:
+        second_subhead_index = None
+        with Path(md_path).open('r+') as md_file:
+            lines = md_file.readlines()
+            count = 0
+            for i, line in enumerate(lines):
+                if line.startswith("## "):
+                    count += 1
+                    if count == 2:
+                        second_subhead_index = i
+                        break
+            if second_subhead_index is None:
+                return
+            md_file.seek(0)
+            md_file.truncate()
+            md_file.writelines(lines[:second_subhead_index])
+
+    except (FileNotFoundError, IsADirectoryError):
+        Logger.debug(f"Unable to open file at {md_path}")
+
+
+def workflow_md_has_task_subheader(md_path: str) -> bool:
+    """
+    Checks if the markdown file has a task subheader
+    :param md_path: Path to markdown file
+    :return: True if the markdown file has a task subheader
+    """
+    try:
+        with open(md_path, 'r') as md_file:
+            md_lines = md_file.readlines()
+            if len(md_lines) == 0:
```

Comment:
lines 177-182 can be out of the `with` block.

Reply:
done